src/modules/rtpengine/rtpengine.c

```diff
@@ -2377,12 +2382,35 @@ static int mos_label_stats_parse(struct minmax_mos_label_stats *mmls)
 static int child_init(int rank)
 {
 	if(!rtpp_set_list)
-		return 0;
+		return 0;	
 
-	/* do not init sockets for PROC_INIT and main process when fork=yes */
-	if(rank == PROC_INIT || (rank == PROC_MAIN && dont_fork == 0)) {
+	/* do not init sockets for PROC_INIT */
+	if(rank == PROC_INIT) {
 		return 0;
-	}
+	}	
+
+	if(rank == PROC_MAIN) {
+		if(rtpengine_dtmf_event_sock.len > 0) {
+			LM_DBG("Register RTPENGINE DTMF WORKER %d\n", mypid);
+			/* fork worker process */
+			mypid = fork_process(PROC_RPC, "RTPENGINE DTMF WORKER", 1);
+			if(mypid < 0) {
+				LM_ERR("failed to fork RTPENGINE DTMF WORKER process %d\n", mypid);
+				return -1;
+			} else if(mypid == 0) {
+				if(cfg_child_init())
+					return -1;
+				/* this will loop forever */
+				rtpengine_dtmf_events_loop();
+			} 
```

Comment:
I suppose there should be an exit() at the end here? Although I'm not too familiar with how the Kamailio internals are handling this

Reply:
Maybe @miconda can comment on that?

Reply:
I think we do not have explicitly `exit` after the functions doing infinite loops in the special child process. But it could be a good practice to kill the process (and by that kamailio instance) if the function that should do infinite loop exits due to some errors before it enters the infinite loop.

Reply:
I don't think it is right to kill the entire kamailio process due to an exit in such function.
I print error logs in case of failure in the function. isn't it enough?
If you prefer that I'll kill the process just let me know and I'll do that.